src/app/content/hooks/locationChange/resolveContent.ts

```diff
@@ -170,20 +179,27 @@ const resolveExternalBookReference = async(
 
 const loadContentReference = async(
   services: AppServices & MiddlewareAPI,
-  book: Book,
+  book: Book | BookWithOSWebData,
   page: ArchivePage,
   reference: ReturnType<typeof getContentPageReferences>[0]
 ) => {
   const targetBook: Book = archiveTreeContainsNode(book.tree, reference.pageUid)
     ? book
     : await resolveExternalBookReference(services, book, page, reference.pageUid);
 
-  return {
-    match: reference.match,
-    params: {
+  const params = hasOSWebData(targetBook)
```

Comment:
looks like `getBookPageUrlAndParams` already exists but you can't use it because you don't have the right page type. we could split out `getUrlParamForBook` in the same way we have `getUrlParamForPageId`